src/packages/hex.gleam

```diff
@@ -3,19 +3,78 @@ import gleam/json
 import gleam/http
 import gleam/result
 import gleam/hackney
+import gleam/list
 import gleam/dynamic.{Decoder} as d
+import time/parse.{parse_iso8601_to_epoch_timestamp}
+import gleam/pgo
+import gleam/string
+import gleam/int
 
 pub type Error {
   HackneyError(hackney.Error)
   JsonError(json.DecodeError)
+  DatabaseError(pgo.QueryError)
+  MiscError(String)
 }
 
-pub fn query() {
+pub fn query(db: pgo.Connection) {
+  try last_scanned = get_last_scanned(db)
+
+  query_all_packages([], 1, last_scanned)
+  |> io.debug
+
+  assert Ok(_) = update_last_scanned(db)
+
+  Ok(1)
+}
+
+const last_scanned_query = "SELECT id, date_trunc('second', scanned_at) FROM previous_hex_api_scan LIMIT 1;"
+
+fn get_last_scanned(db: pgo.Connection) -> Result(Int, Error) {
+  try response =
+    pgo.execute(
+      last_scanned_query,
+      db,
+      [],
+      d.tuple2(
+        d.bool,
+        d.tuple2(d.tuple3(d.int, d.int, d.int), d.tuple3(d.int, d.int, d.int)),
+      ),
+    )
+    |> result.map_error(DatabaseError)
+
+  case response.rows {
+    [first, ..] -> Ok(parse.to_epoch_timetsamp(first.1))
+    _ -> Ok(0)
+  }
+}
+
+const update_last_scanned_query = "UPDATE previous_hex_api_scan SET scanned_at = now();"
+
+fn update_last_scanned(
+  db: pgo.Connection,
+) -> Result(pgo.Returned(d.Dynamic), Error) {
+  pgo.execute(update_last_scanned_query, db, [], d.dynamic)
+  |> result.map_error(DatabaseError)
+}
+
+fn query_all_packages(
+  last_page: List(Package),
+  next_page: Int,
+  last_ran: Int,
+) -> Result(List(Package), Error) {
   let req =
     http.default_req()
     |> http.set_method(http.Get)
+    |> http.prepend_req_header(
+      "User-Agent",
+      "GleamPackages/0.0.1 (Gleam/0.21.0)",
```

Comment:
It would be annoying to update the versions so how about this?
```suggestion
      "GleamPackages",
```

Reply:
Done!